src/Uuid/ModelUuid.php

```diff
@@ -63,56 +63,58 @@ public function id(): string
 		// generate a new ID (to be saved in the content file)
 		$id = static::generate();
 
-		// make sure Kirby has the required permissions
-		// for the update action
-		$kirby = $this->model->kirby();
-		$user  = $kirby->auth()->currentUserFromImpersonation();
-		$kirby->impersonate('kirby');
+		// store the new UUID
+		$this->storeId($id);
 
+		// update the Uri object
+		$this->uri->host($id);
+
+		return $id;
+	}
+
+	/**
+	 * Retrieves the ID string (UUID without scheme) for the model
+	 * from the content file, if it is already stored there
+	 *
+	 * @param \Kirby\Cms\ModelWithContent $model
+	 */
+	public static function retrieveId(Identifiable $model): string|null
+	{
+		return $model->content('default')->get('uuid')->value();
+	}
+
+	/**
+	 * Stores the UUID for the model and makes sure
+	 * to update the content file and content object cache
+	 */
+	protected function storeId(string $id): void
+	{
 		// get the content array from the page
-		$data = $this->model->content()->toArray();
+		$data = $this->model->content('default')->toArray();
 
 		// check for an empty content array
 		// and read content from file again,
 		// just to be sure we don't lose content
 		if (empty($data) === true) {
 			usleep(1000);
-			$data = $this->model->readContent();
+			$data = $this->model->readContent('default');
 		}
 
 		// add the UUID to the content array
 		if (empty($data['uuid']) === true) {
 			$data['uuid'] = $id;
 		}
 
-		// overwrite the content in memory and in the content file;
-		// use the most basic write method to avoid object cloning
-		$this->model->content()->update($data);
-		$this->model->writeContent($data);
-
-		$kirby->impersonate($user);
-
-		// TODO: replace the above in 3.9.0 with
-		// App::instance()->impersonate(
-		// 	'kirby',
-		// 	fn () => $this->model = $this->model->writeContent($data)
-		// );
+		// overwrite the content in memory and in the content file
+		$this->model->content('default')->update($data);
```

Comment:
Isn't this the `else` case when `multilang` is disabled? Or is it needed also in `multilang`?

Reply:
For some reason, the last comments always seem to point to outdated code examples. Could you tell me which line you mean?

Reply:
Ah, now I get it. Let me check.